lib/Backends/BackendUtils.cpp

```diff
@@ -81,6 +74,38 @@ runtime::RuntimeBundle::getSymbolInfo(const Named *v) const {
   return it->second;
 }
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool checkIfOutput(const Placeholder *PH) {
+  for (auto &use : PH->getUsers()) {
+    // Look through the inputs of the PH's users. If an input is overwritten
+    // check if it's the PH, if it is return true.
+    for (unsigned i = 0; i < use.getUser()->getNumInputs(); i++) {
+      if (use.getUser()->isOverwrittenNthInput(i)) {
+        auto input = use.getUser()->getNthInput(i);
+        if (auto overwritten = llvm::dyn_cast<Placeholder>(input)) {
+          if (overwritten == PH) {
+            return true;
+          }
+        }
+      }
+    }
+  }
+  return false;
+}
+
+/// If \p PH is an input placeholder, \returns true.
+bool checkIfInput(const Placeholder *PH) {
```

Comment:
Same here but :%s/output/input.

Reply:
You have a big here. A ph can be the input to a save node. You need to check if this is a destination.

Reply:
I think there’s an additional subtlety here, since a placeholder can be either the input or the overwritten-input of the SaveNode (which is kind of degenerate, but can happen in practice)

Reply:
I think I'm confused. The assumption was if a PH has a single non-SaveNode user it's an input. Is this not the case? Or is it that it can still be considered an input if it's only used by a save node?

Reply:
So the method needs to check that the SaveNode uses the placeholder as a nonOverwritten input?